client/dashboard/sites/site-fields/test/status.tsx

```diff
@@ -0,0 +1,88 @@
+/**
+ * @jest-environment jsdom
+ */
+import { render as testingLibraryRender } from '@testing-library/react';
+import { AuthContext } from '../../../app/auth';
+import { Status } from '../index';
+import type { User, Site } from '../../../data/types';
+
+function render( ui: React.ReactElement ) {
+	return testingLibraryRender(
+		<AuthContext.Provider value={ { user: { ID: 1 } as User } }>{ ui }</AuthContext.Provider>
+	);
+}
+
+describe( '<Status>', () => {
+	test( 'for unlaunched sites, it renders a "Coming soon" link', () => {
+		const site = {
+			slug: 'test.wordpress.com',
+			site_migration: {},
+			launch_status: 'unlaunched',
+			is_private: true,
+		} as Site;
+		const { container, getByRole } = render( <Status site={ site } /> );
+		expect( container.textContent ).toBe( 'Coming soon↗' );
+		expect( getByRole( 'link', { name: /Coming soon/ } ) ).toHaveAttribute(
+			'href',
+			'/home/test.wordpress.com'
+		);
+	} );
+
+	test( 'for launched sites, it renders the site visibility', () => {
+		const site = {
+			site_migration: {},
+			is_private: true,
+		} as Site;
+		const { container } = render( <Status site={ site } /> );
+		expect( container.textContent ).toBe( 'Private' );
+	} );
+
+	test( 'for deleted sites, it renders "Deleted"', () => {
+		const site = {
+			site_migration: {},
+			is_deleted: true,
+		} as Site;
+		const { container } = render( <Status site={ site } /> );
+		expect( container.textContent ).toBe( 'Deleted' );
+	} );
+
+	test( 'for sites still being built via DIFM, it renders "Express service"', () => {
+		const site = {
+			site_migration: {},
+			options: { is_difm_lite_in_progress: true },
+		} as Site;
+		const { container } = render( <Status site={ site } /> );
+		expect( container.textContent ).toBe( 'Express service' );
+	} );
+
+	test( 'for sites with expired plan, it renders "Plan expired"', () => {
+		const site = {
+			site_migration: {},
+			plan: {
+				product_name_short: 'Business',
+				expired: true,
+			},
+		} as Site;
+		const { container } = render( <Status site={ site } /> );
+		expect( container.textContent ).toBe( 'Plan expired' );
+	} );
+
+	test( 'for sites with expired plan, it renders "Plan expired" and a renewal nag for the site owner', () => {
+		const site = {
+			slug: 'test.wordpress.com',
+			site_owner: 1,
```

Comment:
Would it be better to define the userID at the top of the file?

Reply:
Good idea, done!